invenio_records_resources/services/files/schema.py

```diff
@@ -85,7 +85,20 @@ def fields_from_file_obj(self, data, **kwargs):
         if data.file:
             # mandatory fields
             data["storage_class"] = data.file.storage_class
-            data["uri"] = data.file.uri  # TODO: only serialize for external?
+            data["uri"] = data.file.uri
+
+            # If Local -> remove uri as it contains internal file storage info
+            if data["storage_class"] == "L":
```

Comment:
What about using the "enum" instead?
```suggestion
            if data["storage_class"] == TransferType.LOCAL:
```

Reply:
You're right I'll change thx!